web/frontend/src/components/slam-2d-render.vue

```diff
@@ -192,6 +326,28 @@ onUnmounted(() => {
   disposeScene();
 });
 
+watch(() => [props.destVector!.z, props.destVector!.x, props.destExists], async () => {
```

Comment:
1. Does this not throw exceptions if `props.destVector` is undefined?
2. If it does, do we need to do additional checking such as `[props.destVector?.z, props.destVector?.x, props.destExists]`?
3. Could we give this watch function an informative name so it is easier to understand at a glance what it does (moves the robot marker if it exists & moved and removes it if it no longer exists)?
4. Why does this watch function need to be async whereas other watch functions are not?

Reply:
it is async since we have to await makeMarker. 
we have async makeMarker because 
```
const data = await svgLoader.loadAsync(url);
```

Reply:
>Could we give this watch function an informative name so it is easier to understand at a glance what it does (moves the robot marker if it exists & moved and removes it if it no longer exists)?

i didn't know this was a thing. how would you do so?

Reply:
props.destVector is non-nullable per your suggestion

Reply:
Does this work?

```
async updateOrRemoveDestinationMarker() => {
if (props.destVector && props.destExists) {
   ...
}

watch(() => [props.destVector!.z, props.destVector!.x, props.destExists], updateOrRemoveDestinationMarker)
```

Reply:
yep, it does!
thanks :)